src/constraint/encoding.ts

```diff
@@ -138,8 +139,30 @@ export const ENCODING_CONSTRAINTS: EncodingConstraintModel[] = [
       }
       return true;
     }
-  },
-  {
+  },{
+    name: 'scalePropertiesSupportedByScaleType',
+    description: 'Scale properties must be supported by correct scale type',
+    properties: [Property.SCALE, Property.SCALE_TYPE, Property.SCALE_ZERO], // Add more scale properties!
+    allowEnumSpecForProperties: false,
```

Comment:
As I **mentioned** to you in person last week, I would write this constraint in a way that has `allowEnumSpecForProperties` so that we get early pruning rather than pruning after all properties have been sen.